packages/router/src/GenericRouter.php

```diff
@@ -166,6 +164,10 @@ public function isCurrentUri(array|string $action, ...$params): bool
         $currentUri = $this->toUri([$matchedRoute->route->handler->getDeclaringClass(), $matchedRoute->route->handler->getName()]);
 
         foreach ($matchedRoute->params as $key => $value) {
+            if ($value instanceof BackedEnum) {
```

Comment:
I had to refactor some parts of the router, because we now had two dynamic initializers matching a backed enum (a very generic one for routing, and one for the database dialect).

The very generic one matched way too many cases, so I decided to refactor backed enum mapping from an initializer to within the router, which makes more sense anyway.